src/components/ChickenPage/ChickenPage.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import { useLocation } from "react-router-dom";
+import style from "./ChickenPage.module.scss";
+
+const ChickenPage = () => {
+  const { state } = useLocation();
+  const { menu, id, image, price, size, quantity: qty } = state;
+  const [quantity, setQuantity] = useState(qty);
+  const [totalPrice, setTotalPrice] = useState(""); //SUNNY
+  const [orderInput, setOrderInput] = useState("");
+  console.log("state", state);
+  const handleClick = (e) => {
+    e.preventDefault();
+    setQuantity(quantity - orderInput);
+    setTotalPrice(orderInput * price);
+    setOrderInput("");
+  };
```

Comment:
``` 
const handleClick = (e) => {
        e.preventDefault();
        setQuantity(quantity - orderInput);
        setOrderInput("");
    };

    useEffect(() => {
        setTotalPrice((qty - quantity) * price);
    }, [quantity]);
   ```
   we need ot know the most recent quantity to calculate total price -> if we have setQuantity and setProce inside the same function, we won't have access to updated quantity. Instead we can use useEffect - it is guaranteed that it will run after quantity is updated